File: PageantManager.Business/Entities/CostumeGarment.cs
namespace PageantManager.Business.Entities
{
    public class CostumeGarment
    {
        public int CostumeGarmentId { get; set; }
        public int CostumeId { get; set; }
        public int GarmentTypeId { get; set; }

        public Costume Costume { get; set; }
        public GarmentType GarmentType { get; set; }
    }
}

File: PageantManager.Web/src/app/shared/models/costume-garment.ts
import { GarmentType } from './garment-type';

export class CostumeGarment {
  costumeGarmentId: number;
  costumeId: number;
  garmentTypeId: number;
  garmentType: GarmentType;

  public constructor(init?: Partial<CostumeGarment>) {
    Object.assign(this, init);
  }
}
